src/logic.js

```diff
@@ -0,0 +1,149 @@
+import Project from './projects';
+import Todo from './todo';
+
+const STORAGE_KEY = 'project.list';
+const projectList = JSON.parse(localStorage.getItem(STORAGE_KEY)) || [];
+
+function save() {
+  localStorage.setItem(STORAGE_KEY, JSON.stringify(projectList));
+}
+
+const createProject = (title) => {
+  const newProject = Project(title);
+  projectList.push(newProject);
+  save();
+};
+
+const deleteTodo = (project, id) => {
+  const deleteBtn = document.getElementById(`delete-btn-${id}`);
+  const todoItem = document.getElementById(`todo-item-${id}`);
+  deleteBtn.addEventListener('click', () => {
+    todoItem.remove();
+    project.todoList.splice(id, 1);
+    save();
+  });
+};
+
+const removeProject = () => {
+  const getDeleteBtn = document.querySelector('.delete-btn');
+  getDeleteBtn.addEventListener('click', () => {
+    const deleteProject = document.querySelector('.active-project');
+    deleteProject.remove();
+    for (let i = 0; i < projectList.length; i += 1) {
+      if (projectList[i].title === deleteProject.textContent) {
+        projectList.splice(i, 1);
+        const todoItemContainer = document.querySelector('.todo-item-container');
+        while (todoItemContainer.firstChild) {
+          todoItemContainer.removeChild(todoItemContainer.firstChild);
+        }
+        const newTodoContainer = document.querySelector('.new-todo-container');
+        newTodoContainer.style.display = 'none';
+        save();
+      }
+    }
+  });
+};
+
+const createTodo = (title, description, dueDate, priority, project) => {
+  const newTodo = Todo(title, description, dueDate, priority);
+  project.todoList.push(newTodo);
+  save();
+};
+
+const editTodo = (project, id) => {
+  const todoTitle = document.getElementById(`todo-item-title-${id}`);
+  todoTitle.addEventListener('click', () => {
+    const todoItemLeft = document.getElementById(`todo-item-left-${id}`);
+    const todoEditTitle = document.createElement('input');
+    todoEditTitle.classList.add('edit-todo-title');
+    todoEditTitle.value = todoTitle.textContent;
+    todoTitle.textContent = '';
+    todoItemLeft.appendChild(todoEditTitle);
+    todoEditTitle.addEventListener('keydown', (e) => {
+      if (e.keyCode === 13) {
+        if (todoEditTitle.value !== '') {
+          todoTitle.textContent = todoEditTitle.value;
+          todoEditTitle.remove();
+          save();
+        }
+      }
+    });
+  });
+
```

Comment:
this file is called `logic.js` but there's a lot DOM manipulation inside........

Reply:
Validation added and move DOM manipulation from logic